homeassistant/components/nextcloud/binary_sensor.py

```diff
@@ -0,0 +1,52 @@
+"""Summary binary data from Nextcoud."""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+
+from . import BINARY_SENSORS
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "nextcloud"
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None) -> None:
+    """Set up the Nextcloud sensors."""
+    binary_sensors = []
+    for name in hass.data[DOMAIN]:
+        if name in BINARY_SENSORS:
+            binary_sensors.append(NextcloudBinarySensor(name))
+    add_entities(binary_sensors, True)
+
+
+class NextcloudBinarySensor(BinarySensorDevice):
+    """Represents a Nextcloud binary sensor."""
+
+    def __init__(self, item):
+        """Initialize the Nextcloud binary sensor."""
+        self._name = item
+        self._is_on = None
+
+    @property
+    def icon(self):
+        """Return the icon for this binary sensor."""
+        return "mdi:cloud"
+
+    @property
+    def name(self):
+        """Return the name for this binary sensor."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if the binary sensor is on."""
+        return self._is_on == "yes"
+
+    @property
+    def unique_id(self):
+        """Return the unique ID for this binary sensor."""
+        return self._name
```

Comment:
If the integration is extended in the future to support more than one nextcloud instance, this unique id won't be unique anymore. It would be good to make it future proof.

Is there some instance id we can use that is unique and stable?

https://developers.home-assistant.io/docs/entity_registry_index#unique-id-requirements

Reply:
Good point. I'll check, but I'm not sure there is any unique information returned by the api, but the url to the nextcloud instance would at least be unique.

Reply:
Yes:
```
CONFIG_SCHEMA = {
    DOMAIN: vol.Schema(
        {
            vol.Required(CONF_URL): cv.url,
            vol.Required(CONF_USERNAME): cv.string,
            vol.Required(CONF_PASSWORD): cv.string,
            vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL): cv.time_period,
        }
    )
}
```
...
```
conf = config[DOMAIN]
```

Reply:
Would it be too ugly to use the nextcloud instance url as part of the unique_id for the sensors/binary_sensors? 

Something like: `https://nextcloud.myurl.com#nextcloud_num_files`

Reply:
If it's unique and stable its ok.